compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java

```diff
@@ -68,16 +68,23 @@ public enum DiagnosticsCodes {
             ERROR),
     PERSIST_121("PERSIST_121", "unsupported features: {0} type is not supported",
             ERROR),
+    PERSIST_122("PERSIST_122", "Contains composite value: could not infer the reference as " +
+            "the key or uniqueConstrains does not contain single value", ERROR),
+    PERSIST_123("PERSIST_123", "missing key and uniqueConstrains fields: " +
+            "could not infer the reference as key and uniqueConstrains fields are not in the entity annotation", ERROR),
 
+    PERSIST_124("PERSIST_124", "empty value: Could not infer the reference as " +
+            "the uniqueConstraints has an empty value", ERROR),
```

Comment:
references are always inferred from primary keys. why we need to check the unique Constraints

Reply:
Removed in [61202be](https://github.com/ballerina-platform/module-ballerina-persist/pull/67/commits/61202bea12734b99c52aa1c51bb576e0dc4151b7)